sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableAsyncBatch.java

```diff
@@ -0,0 +1,385 @@
+     */
+    public synchronized List<BatchOperation> getOperations() {
+        return Collections.unmodifiableList(this.operations);
+    }
+
+    /**
+     * Executes all operations within The batch inside a transaction. When the call completes, either all operations in
+     * the batch will succeed, or if a failure occurs, all operations in the batch will be rolled back.
+     *
+     * @return A reactive result containing a list of sub-responses for each operation in the batch.
+     * @throws TableServiceErrorException if any operation within the batch fails. See the documentation for the client
+     *                                    methods in {@link TableAsyncClient} to understand the conditions that may
+     *                                    cause a given operation to fail.
+     * @throws IllegalStateException if no operations have been added to the batch.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public synchronized Mono<List<BatchOperationResponse>> submitTransaction() {
+        return submitTransactionWithResponse().flatMap(response -> Mono.justOrEmpty(response.getValue()));
+    }
+
+    /**
+     * Executes all operations within The batch inside a transaction. When the call completes, either all operations in
+     * the batch will succeed, or if a failure occurs, all operations in the batch will be rolled back.
+     *
+     * @return A reactive result containing the HTTP response produced for the batch itself. The response's value will
+     * contain a list of sub-responses for each operation in the batch.
+     * @throws TableServiceErrorException if any operation within the batch fails. See the documentation for the client
+     *                                    methods in {@link TableAsyncClient} to understand the conditions that may
+     *                                    cause a given operation to fail.
+     * @throws IllegalStateException if no operations have been added to the batch.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public synchronized Mono<Response<List<BatchOperationResponse>>> submitTransactionWithResponse() {
+        return withContext(this::submitTransactionWithResponse);
+    }
+
+    synchronized Mono<Response<List<BatchOperationResponse>>> submitTransactionWithResponse(Context context) {
+        this.frozen = true;
+        context = context == null ? Context.NONE : context;
+
+        if (operations.size() == 0) {
+            throw logger.logExceptionAsError(new IllegalStateException("A batch must contain at least one operation."));
+        }
+
+        final BatchRequestBody body = new BatchRequestBody();
+        Flux.fromIterable(operations)
+            .flatMapSequential(op -> op.prepareRequest(operationClient))
+            .zipWith(Flux.fromIterable(operations))
+            .doOnNext(pair -> body.addChangeOperation(new BatchSubRequest(pair.getT2(), pair.getT1())))
+            .blockLast();
```

Comment:
Shouldn't block here

Reply:
Indeed, I would love to find a non-blocking way to do this. The only async operation is the `op.prepareRequest` step, which "executes" the request using the new `NullHttpClient` which doesn't actually send the request on the wire, it just captures the request so we can read out its method / headers / body and stuff them into the multipart body later on. So the blocking call here isn't as bad as it looks, but it still would be ideal to find a way to remove it.